javagen/src/main/java/com/azure/autorest/mapper/ClientParameterMapper.java

```diff
@@ -79,6 +82,12 @@ public ClientMethodParameter map(Parameter parameter) {
         if (description == null || description.isEmpty()) {
             description = String.format("The %s parameter", name);
         }
+        // add allowed enum values
+        if (settings.isDataPlaneClient() && parameter.getProtocol().getHttp().getIn() != RequestParameterLocation.BODY && Mappers.getSchemaMapper().map(parameter.getSchema()) instanceof EnumType) {
+            description += ". ";
```

Comment:
What if description already ends with "."?

Can it be moved to buildAllowedEnumValues?

Why you need to map it `Mappers.getSchemaMapper().map(parameter.getSchema())`? Can you move all these to buildAllowedEnumValues?